server/app/database/model_custom_set.py

```diff
@@ -17,7 +20,54 @@ class ModelCustomSet(Base):
     name = Column("name", String)
     description = Column("description", String)
     owner_id = Column(UUID(as_uuid=True), ForeignKey("user.uuid"), index=True)
-    created_at = Column("creation_date", DateTime)
+    created_at = Column("creation_date", DateTime, default=datetime.now)
+    last_modified = Column("last_modified", DateTime, default=datetime.now, index=True)
     level = Column("level", Integer)
-    equipped_items = relationship("ModelItemSlot", secondary="equipped_item")
+    equipped_items = relationship(
+        "ModelEquippedItem", backref="custom_set", lazy="dynamic"
+    )
     stats = relationship("ModelCustomSetStat", cascade="all, delete-orphan")
+
+    def find_empty_item_slot(self, item_type):
+        eligible_item_slots = item_type.eligible_item_slots
+        for item_slot in eligible_item_slots:
+            equipped_item = ModelEquippedItem.query.filter_by(
+                custom_set_id=self.uuid, item_slot_id=item_slot.uuid
+            ).one_or_none()
+            if not equipped_item:
+                return item_slot
+        return None
+
+    def equip_item(self, item_id, item_slot_id):

```

Comment:
In what cases do you not provide it an item slot?

Reply:
Item slot is required now